### PR TITLE
Add module sync actions

### PR DESCRIPTION
**Pull request by NaughtyChas**
Well Github Actions can only be triggered on main branch, merging actions to main for testing.
Repository secret has added so token can be passed securely.

I have created a backup branch of main called 'main-backup' if anything goes wrong.

### CODE REVIEW

File: .github/workflows/submoduleNotifyParent.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - main
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with: 
+          repository: org/parent_repository
```
Comment (CloneWith) [inline .github/workflows/submoduleNotifyParent.yml:15]: This seems like a placeholder, should change to the proper repo name

```suggestion
          repository: osu-atri/osu-dictionary
```

File: .github/workflows/submoduleNotifyParent.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          token: ${{ secrets.PRIVATE_TOKEN_GITHUB }}
+          submodules: true
+
+      - name: Pull & update submodules recursively
```
Comment (CloneWith) [inline .github/workflows/submoduleNotifyParent.yml:19]: When running to this step, need to make sure if we're in the correct directory, or to use `cd` to change it.

File: .github/workflows/submoduleNotifyParent.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        run: |
+          git submodule update --init --recursive
+          git submodule update --recursive --remote
+
+      - name: Commit
```
Comment (CloneWith) [inline .github/workflows/submoduleNotifyParent.yml:24]: If we are going to commit the updated result directly to the `main` branch, will it be blocked? You know we have a main branch protection rule here, and may need to make that excluded for GitHub Actions.

File: .github/workflows/submoduleNotifyParent.yml (hunk 1 above)
Comment (NaughtyChas) [reply .github/workflows/submoduleNotifyParent.yml]: Just forgot to change that, thx

File: .github/workflows/submoduleNotifyParent.yml (hunk 2 above)
Comment (NaughtyChas) [reply .github/workflows/submoduleNotifyParent.yml:19]: `actions/checkout@v2` action checks out the repository and its submodules into the default working directory, which is the root of the repository. Therefore these commands should run in the correct directory without needing to use `cd`.

Comment (CloneWith) [reply .github/workflows/submoduleNotifyParent.yml:19]: Just had a look at `osu-wiki` CI workflow, and it turns out that this concern is unnecessary. Actions should have made this for us.

File: .github/workflows/submoduleNotifyParent.yml (hunk 3 above)
Comment (NaughtyChas) [reply .github/workflows/submoduleNotifyParent.yml:30]: Using automatically pulling PRs instead, available in [this commit](https://github.com/osu-atri/lazer-wiki/pull/3/commits/35de0630e35da7bda679c8206ebe6456e173cfc2).

**Review by CloneWith — APPROVED** [review]

**Comment by NaughtyChas** [comment]
Merge and see if it works now. Further discussions is allowed under this PR after merge.

**Comment by NaughtyChas** [comment]
No enough access for R/W for the Fine-Grained secret. Looking for a fix...

**Pull request merged**